Sources/BuildServerProtocol/BuildTargets.swift

```diff
@@ -207,3 +207,44 @@ public struct OutputsItem: Codable, Hashable {
   /// The output paths for sources that belong to this build target.
   public var outputPaths: [URL]
 }
+
+/// The build target changed notification is sent from the server to the client
+/// to signal a change in a build target. The server communicates during the
+/// initialize handshake whether this method is supported or not.
+public struct BuildTargetsChangedNotification: NotificationType {
+  public static let method: String = "buildTarget/didChange"
+
+  public var changes: [BuildTargetEvent]
+
+  public init(changes: [BuildTargetEvent]) {
+    self.changes = changes
+  }
+}
+
+public struct BuildTargetEvent: Codable, Hashable {
+  /// The identifier for the changed build target.
+  public var target: BuildTargetIdentifier
+
+  /// The kind of change for this build target.
+  public var kind: BuildTargetEventKind
```

Comment:
The spec says this should be optional (https://github.com/build-server-protocol/build-server-protocol/blob/master/docs/specification.md#build-target-changed-notification), although I'm not sure what the semantics of `nil` are here. Any thoughts?

Reply:
Yes, that is rather confusing. This value is nice as it allows for not invalidating the build target graph when a new target is created or deleted, you can just request the updated sources and outputs for the changed module. I suppose if not set the language server has to just refresh everything. I'll update this to be optional.